.gitlab-ci.yml

```diff
@@ -61,20 +63,43 @@ before_script:
       - ${YARN_CACHE_FOLDER}
     policy: pull-push
 
+# ==================== rules ==================== #
+.preview_rules: &preview_rules
+  rules:
+    - if: $CI_COMMIT_BRANCH != $CI_DEFAULT_BRANCH && $CI_COMMIT_BRANCH =~ $PREVIEW_PATTERN
+      when: on_success
+
+.preview_manual_rules: &preview_manual_rules
+  rules:
+    - if: $CI_COMMIT_BRANCH != $CI_DEFAULT_BRANCH && $CI_COMMIT_BRANCH =~ $PREVIEW_PATTERN
+      when: manual
+
+.live_rules: &live_rules
+  rules:
+    - if: $CI_COMMIT_REF_NAME == "master"
+      when: on_success
+
+.live_schedule_rules: &live_schedule_rules
+  rules:
+    - if: $CI_COMMIT_REF_NAME == "master" && $CI_PIPELINE_SOURCE == "schedule"
+      when: on_success
 # ================== templates ================== #
 .base_template: &base_template
   image: public.ecr.aws/x2b9z2t7/webops/site-build:latest
   tags:
     - "runner:main"
-  except:
-    - tags
-  only:
-    - branches
+  rules:
+    - if: $CI_PIPELINE_SOURCE == "tags"
```

Comment:
Should the value for For  the $CI_PIPELINE_SOURCE be `push` or  instead use `if: $CI_COMMIT_TAG`? https://docs.gitlab.com/ee/ci/jobs/job_control.html#common-if-clauses-for-rules

Reply:
Good catch, the variable should be the tag one here, with a similar `!= null` to ensure it does not run on tagging.

Reply:
i didnt see 'tags' as a possible value for the PIPELINE SOURCE

Reply:
Yea it wasn't it was a mistake on my part, the commit tag you listed only exists when it's a tag job, which we want to prevent so by check for it not equaling null we will be able to prevent the run.